### DISCUSSION TITLE
Функционал

### DISCUSSION THREAD

**Discussion by enslit** [ezinber/pomo-project#3] [General]
**Предложения по функционалу:** 
1. Регистрация/авторизация? 
       _Для этого нужен небольшой бек, либо можно сделать на firebase_
2. Настройки таймера
     - время помидора _(по умолчанию 25 минут)_
     - время короткого перерыва _(по умолчанию 5 минут)_
     - время длинного перерыва _(по умолчанию 15 минут)_
3. Сохранение/чтение настроек таймера в/из local storage
4. Создание задачи с указанием планируемого количества помидор
5. Выбор активной задачи в счетчике. По умолчанию по порядку, но можно сделать возможность изменить порядок

**Comment by ezinber** [comment]
1. Регистрацию, я думаю, мы прикрутим позже, как подтянемся в знании бэкенда, а сейчас есть идея реализовать сохранение текущих настроек в гет-параметрах. Подобная штука есть в поисковике DuckDuckGo - сейчас изучаю эту тему поподробней, если слишком муторно, обойдёмся пока вообще без этого
2. Да, сначала можно сделать жёстко заданные параметры, а кастомизацию времени оставить на допиливание
3. -//-
4. Задавать "лимит" помидорин - это интересно. Можно сделать это как дневной рекомендуемый лимит. На задаче сделать иконку "🍅" и указывать общее количество завершённых за всё время, типа "🍅 х 3", ну или цифрой внутри помидорки
5. Я представляю клик по задаче либо что-то типа кнопки play на карточке задачи, которая запускает таймер и показывает заголовок текущей задачи. По порядку - это типа по достижении лимита на одной задаче таймер сам переключается на другую?

**Comment by enslit** [comment]
> 1. Регистрацию, я думаю, мы прикрутим позже, как подтянемся в знании бэкенда, а сейчас есть идея реализовать сохранение текущих настроек в гет-параметрах. Подобная штука есть в поисковике DuckDuckGo - сейчас изучаю эту тему поподробней, если слишком муторно, обойдёмся пока вообще без этого
> 
Держим на горизонте

> 4. Задавать "лимит" помидорин - это интересно. Можно сделать это как дневной рекомендуемый лимит. На задаче сделать иконку "🍅" и указывать общее количество завершённых за всё время, типа "🍅 х 3", ну или цифрой внутри помидорки
> 
Я тут думал что-то вроде план-факта сделать 

> 5. Я представляю клик по задаче либо что-то типа кнопки play на карточке задачи, которая запускает таймер и показывает заголовок текущей задачи. По порядку - это типа по достижении лимита на одной задаче таймер сам переключается на другую?
> 
Я думал об общей кнопке "старт" по нажатию на которую, время тикать начинает по задаче, первой в списке. А по достижении лимита, ничего не делать с задачей, просто продолжать добавлять помидоры, чтобы было видно, что оценка задачи получилась не точной. Завершать задачи только вручную

**Reply by ezinber** [reply]
> > 1. Задавать "лимит" помидорин - это интересно. Можно сделать это как дневной рекомендуемый лимит. На задаче сделать иконку "tomato" и указывать общее количество завершённых за всё время, типа "tomato х 3", ну или цифрой внутри помидорки
> 
> Я тут думал что-то вроде план-факта сделать

Да, наверное для более продвинутых организаторов своего времени штука нужная. Я просто заранее вряд-ли решу, сколько помидоров может понадобиться, поэтому ограничиваю себя суточным лимитом на задачу

> > 1. Я представляю клик по задаче либо что-то типа кнопки play на карточке задачи, которая запускает таймер и показывает заголовок текущей задачи. По порядку - это типа по достижении лимита на одной задаче таймер сам переключается на другую?
> 
> Я думал об общей кнопке "старт" по нажатию на которую, время тикать начинает по задаче, первой в списке. А по достижении лимита, ничего не делать с задачей, просто продолжать добавлять помидоры, чтобы было видно, что оценка задачи получилась не точной. Завершать задачи только вручную

А что если на таймер можно щёлкать без фокуса на задаче? Типа я открыл сайт, у меня сохранены задачи, но сейчас вот решил поугарать над мемчиками и по-привычке отвёл на это один помидор. Создавать задачу - лишнее, а так - просто таймер запустил

**Reply by enslit** [reply]
Наверное я слишком заморачиваюсь) Давай тогда пока сделаем сам по себе таймер и список задач, которые могут запускать таймер? А помидоры тогда считать общее количество за день

**Reply by ezinber** [reply]
Согласен, начнём с таймера, а там может и другие подсоединятся

**Comment by nyamnyannyak** [comment]
Ребята, привет! Не знаю, какую взять себе задачку. Есть что-то для меня?)

**Reply by ezinber** [reply]
Привет! Да мы пока устанавливаем связи всего со всем. Раскидать задачи - это не так-то просто для новичков. Надо накидывать дизайн, из функционала пока сам таймер нужен. И там тестовый пулл-реквест висит, ожидает твоего ревью
